core/positions/engine.go

```diff
@@ -135,7 +180,7 @@ func (e *Engine) RegisterOrder(ctx context.Context, order *types.Order) *MarketP
 	}
 
```

Comment:
Well, we set it in the constructor in any case? I'm not sure an explicit Panic from us is more useful than the one we would hit because of a nil pointer haha?